pkg/apis/provisioning.cattle.io/v1/cluster_types.go

```diff
@@ -107,49 +315,164 @@ type RKEMachinePoolRollingUpdate struct {
+	// the agent deployment.
+	// +optional
+	AppendTolerations []v1.Toleration `json:"appendTolerations,omitempty"`
+
+	// OverrideAffinity is an affinity that will be used to override the
+	// agent deployment's affinity.
+	// +optional
+	OverrideAffinity *v1.Affinity `json:"overrideAffinity,omitempty"`
+
+	// OverrideResourceRequirements defines the limits, requests, and
+	// claims of compute resources for a given container.
+	// +optional
+	OverrideResourceRequirements *v1.ResourceRequirements `json:"overrideResourceRequirements,omitempty"`
+
+	// SchedulingCustomization is an optional configuration that will be
+	// used to override the agent deployment's scheduling customization.
+	// +optional
+	SchedulingCustomization *AgentSchedulingCustomization `json:"schedulingCustomization,omitempty"`
 }
 
 type AgentSchedulingCustomization struct {
-	PriorityClass       *PriorityClassSpec       `json:"priorityClass,omitempty"`
+	// PriorityClass is the configuration for the priority class associated
+	// with the agent deployment.
+	// +optional
+	PriorityClass *PriorityClassSpec `json:"priorityClass,omitempty"`
+
+	// PodDisruptionBudget is the configuration for the pod disruption budget
+	// associated with the agent deployment.
+	// +optional
 	PodDisruptionBudget *PodDisruptionBudgetSpec `json:"podDisruptionBudget,omitempty"`
 }
 
 type PriorityClassSpec struct {
-	Value            int                  `json:"value,omitempty"`
+	// Value represents the integer value of this priority class.
+	// This is the actual priority value that pods receive when their
+	// associated deployment specifies name of this class in the spec.
+	// +kubebuilder:validation:Maximum=1000000000
+	// +kubebuilder:validation:Minimum=-1000000000
+	// +optional
+	Value int `json:"value,omitempty"`
+
+	// PreemptionPolicy describes a policy for if/when to preempt a pod.
+	// +kubebuilder:validation:Enum=PreemptLowerPriority;Never
+	// +optional
 	PreemptionPolicy *v1.PreemptionPolicy `json:"preemptionPolicy,omitempty"`
 }
 
+// PodDisruptionBudgetSpec is the spec for the desired pod disruption budget for the cluster.
+// +kubebuilder:validation:XValidation:rule="(has(self.minAvailable) && self.minAvailable != \"0\" && (!has(self.maxUnavailable) || self.maxUnavailable == \"0\")) || (has(self.maxUnavailable) && self.maxUnavailable != \"0\" && (!has(self.minAvailable) || self.minAvailable == \"0\"))",message="both minAvailable and maxUnavailable cannot be set to a non-zero value, at least one must be omitted or set to zero"
 type PodDisruptionBudgetSpec struct {
-	MinAvailable   string `json:"minAvailable,omitempty"`
+	// An eviction is allowed if at least "minAvailable" will still be
+	// available after the eviction, i.e. even in the absence of the evicted
+	// pod.
+	// One can prevent all voluntary evictions by specifying "100%".
+	// +kubebuilder:validation:Pattern="^((([1-9]|[1-9][0-9]|100)%)|([1-9][0-9]*|0))$"
```

Comment:
If we want to allow `""` as we probably currently do, this patter would block it. Same for `MaxUnavailable`.